### PR TITLE
build library with hdf5 1 12 0

### LINKED ISSUE #1628: build netcdf with hdf5-1.12.0
Well hdf5-1.12.0 beta has been released! How exciting.

### PR DESCRIPTION
**Pull request by edwardhartnett**
Fixes #1628 

This is the work of @brtnfld, updated to the current master. Thanks Scott!

This is failing CMake builds for me but that may be something wrong with my newly-rebuilt CI system.

### CODE REVIEW

**Comment by edwardhartnett** [comment]
@WardF I believe this is ready to merge...

**Comment by edwardhartnett** [comment]
@WardF this is ready to merge, and should be merged before the next release...

**Comment by WardF** [comment]
Reviewing now, thanks for your patience!

File: h5_test/tst_h_atts3.c [hunk 1]
```diff
@@ -46,7 +46,11 @@ main()
       hid_t file_typeid1[NUM_OBJ], native_typeid1[NUM_OBJ];
       hid_t file_typeid2, native_typeid2;
       hsize_t num_obj;
+#if H5_VERSION_GE(1,12,0)
```
Comment (WardF) [inline h5_test/tst_h_atts3.c]: Unrelated to this PR, but a note. @DennisHeimbigner, this H5_VERSION_GE macro seems useful, I wonder if we should add a netcdf equivalent? We can discuss later, mentioning it now before I forget (again).

Comment (DennisHeimbigner) [reply h5_test/tst_h_atts3.c]: You mean something like NC_VERSION_GE?

**Pull request merged**